packages/jest-runner/src/index.ts

```diff
@@ -134,6 +134,9 @@ class TestRunner {
               Array.from(this._context.changedFiles),
           },
           globalConfig: this._globalConfig,
+          moduleMapUniqueID: watcher.isWatchMode()
+            ? test.context.moduleMap.uniqueID
+            : null,
           path: test.path,
           serializableModuleMap: watcher.isWatchMode()
             ? test.context.moduleMap.toJSON()
```

Comment:
@SimenB I've been thinking about this and I'm pretty sure we're still paying a serialization cost to send this to the worker for each test, just much less of one. Is this part of the public interface? Can I change it without it being considered a breaking change?

Can always explore in further PRs since this one is obviously an improvement, but it would be nice if I was able to move the logic up into the main thread and not even send the haste map if it wasn't new. I'm not sure if it is a breaking change.

Reply:
I'm not sure... That's more of a question for @rubennorte or @mjesun 

The less we have to send across ipc the better, so if we can change it I'm all for it 🙂

Reply:
It looks like external runners include both the worker and the orchestrator, so changing what we pass to the worker here should be safely in the realm of a non-breaking-change!

Working to further optimize this.

@rubennorte @mjesun Let me know if I'm incorrect.

Reply:
@SimenB Feel free to re-review, I've refactored this to approach it from a different angle and it's now even faster. On Jest's test suite, before Watch mode was a 6%~ performance cost and now it's about 1%~.

The refactor matters even more for larger haste maps, like Facebook's.